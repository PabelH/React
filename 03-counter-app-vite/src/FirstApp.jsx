// function App() {
//     return <h1>Hello Pabello</h1>
// }

//using rafc
// import React from 'react'

const FirstApp = () => {
  return (
    <div>Hello Pabelon </div>
  )
}

export default FirstApp



// export default App;
