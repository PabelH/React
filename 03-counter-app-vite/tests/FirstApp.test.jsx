import { render } from "@testing-library/react";
import FirstApp from "../src/FirstApp";


describe('Test in <FirstApp />', () => {
    test('this test must match the snapshot"',()=>{

        const title = 'Hi MF'
        const {container} = render(<FirstApp title={ title }/>);
        //console.log(container);
        expect( container ).toMatchSnapshot();


        
    });
    test('this test must render the title in a h1 tag"',()=>{

        const title = 'Hi MF'
        const {container, getByText} = render(<FirstApp title={ title }/>);
        expect( getByText(title) ).toBeTruthy();
        const h1 = container.querySelector('h1');
        // expect( h1.innerHTML ).toBe(title);
        expect( h1.innerHTML ).toContain(title);



        
    });
  
});