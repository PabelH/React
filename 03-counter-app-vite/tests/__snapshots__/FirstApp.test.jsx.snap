// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <FirstApp /> this test must match the snapshot" 1`] = `
<div>
  <h1>
     
    Hi MF
     
  </h1>
  <h2>
    No Subtitle
  </h2>
  <h2>
    Pabelon
  </h2>
</div>
`;
